packages/form-layout/src/vaadin-form-item.js

```diff
@@ -164,14 +165,84 @@ class FormItem extends ThemableMixin(PolymerElement) {
     super();
     this.__updateInvalidState = this.__updateInvalidState.bind(this);
     this.__contentFieldObserver = new MutationObserver(() => this.__updateRequiredState(this.__contentField.required));
+
+    // Ensure every instance has unique ID
+    const uniqueId = (FormItem._uniqueLabelId = 1 + FormItem._uniqueLabelId || 0);
+    this.__labelId = `label-${this.localName}-${uniqueId}`;
+  }
+
+  /**
+   * An array of field nodes of the content slot.
+   *
+   * @return {HTMLElement[]}
+   * @private
+   */
+  get __fieldNodes() {
+    return this.$.contentSlot.assignedElements();
```

Comment:
Should probably move [the filtering for actual field nodes](https://github.com/vaadin/web-components/pull/3088/files#diff-b39d7090b859fd2c9e9cdc8703ccb7ce4ebbf88dcbf1ea14c0914bf540d11513R298) here. The current `__fieldNodes` getter returns all slotted elements from the content slot, not just fields.

Reply:
Hmm, good point. The field is only an element that has `checkValidity` or `validate`, sounds correct.

Reply:
Refactored, please take a look.